### PR TITLE
Add state and description codes: Part 1

### PR DESCRIPTION
**Pull request by 0xB10C**
This PR adds the foundation and groundwork for the state and description codes. The handling by the Supervisor and the Middleware will come in part 2. 

~_I leave this as a draft until https://github.com/digitalbitbox/bitbox02-api-go/pull/17 is merged_~

### CODE REVIEW

File: tools/bbbsupervisor/go.mod [hunk 1]
```diff
@@ -3,7 +3,9 @@ module github.com/digitalbitbox/bitbox-base/tools/bbbsupervisor
...
 require (
-	github.com/gomodule/redigo v2.0.0+incompatible
-	github.com/stretchr/testify v1.4.0 // indirect
-	github.com/tidwall/gjson v1.3.2
+	github.com/digitalbitbox/bitbox-base/middleware v0.0.0-20191203134058-7c461ef0407d
+	github.com/digitalbitbox/bitbox02-api-go v0.0.0-20191122093321-5bacb3c08094
+	github.com/tidwall/gjson v1.3.4
 )
+
+replace github.com/digitalbitbox/bitbox02-api-go => /home/b10c/shift/bitbox02-api-go // TODO: remove before merge
```
Comment (0xB10C) [inline tools/bbbsupervisor/go.mod]: Will remove when https://github.com/digitalbitbox/bitbox02-api-go/pull/17 is merged.

File: middleware/src/logtags/logtags.go [hunk 2]
```diff
@@ -0,0 +1,36 @@
+package logtags
+
+// These LogTags are shared logtags between the Middleware and the Supervisor.
+// The supervisor watches these and triggers the correspointing handler.
```
Comment (Stadicus) [inline middleware/src/logtags/logtags.go]: "corresponding"

**Comment by 0xB10C** [comment]
fix typo `correspointing` -> `corresponding`

File: middleware/src/redis/keys.go [hunk 3]
```diff
@@ -19,4 +19,5 @@ const (
 	MiddlewareAuth        BaseRedisKey = "middleware:auth"
 	BaseSetupDone         BaseRedisKey = "base:setup"
 	BaseSSHDPasswordLogin BaseRedisKey = "base:sshd:passwordlogin"
+	BitcoindIDB           BaseRedisKey = "bitcoind:ibd"
```
Comment (Tomasvrba) [inline middleware/src/redis/keys.go]: typo `IBD` not `IDB`

File: tools/bbbsupervisor/supervisor/util.go [hunk 4]
```diff
@@ -5,13 +5,15 @@ import (
...
+
+	"github.com/digitalbitbox/bitbox-base/middleware/src/redis"
 )
 
 // unsetClearnetIDB unsets (0 - download blocks over Tor) the ibdClearnetRedisKey if set.
 // The key can only be set back to 1 (download blocks over clearnet) via RPC.
 func (s *Supervisor) unsetClearnetIDB() (err error) {
 	const ibdClearnetRedisKey string = "bitcoind:ibd-clearnet"
-	isIBDClearnet, err := s.redis.GetInt(ibdClearnetRedisKey)
+	isIBDClearnet, err := s.redis.GetInt(redis.BitcoindIDB)
```
Comment (Tomasvrba) [inline tools/bbbsupervisor/supervisor/util.go]: Same typo `IDB`

File: tools/bbbsupervisor/systemstate/systemstate.go [hunk 5]
```diff
@@ -0,0 +1,30 @@
...
+	messages.BitBoxBaseHeartbeatRequest_DOWNLOAD_UPDATE:       1400,
+	messages.BitBoxBaseHeartbeatRequest_UPDATE_FAILED:         2300,
+	messages.BitBoxBaseHeartbeatRequest_NO_NETWORK_CONNECTION: 2400,
+	messages.BitBoxBaseHeartbeatRequest_REBOOT:                2500,
+	messages.BitBoxBaseHeartbeatRequest_SHUTDOWN:              2510,
+	messages.BitBoxBaseHeartbeatRequest_OUT_OF_DISK_SPACE:     3400,
+	messages.BitBoxBaseHeartbeatRequest_REDIS_ERROR:           3900,
+}
+
+// MapDescriptionCodeStateCode defines the corresponding the StateCode for a DescriptionCode.
```
Comment (Tomasvrba) [inline tools/bbbsupervisor/systemstate/systemstate.go]: This description has a little strange wording. Maybe something like:
> MapDescriptionCodeStateCode maps DescriptionCodes to the corresponding StateCodes ?

File: tools/bbbsupervisor/supervisor/util.go (hunk 4 above)
Comment (0xB10C) [reply tools/bbbsupervisor/supervisor/util.go]: Actually there is a way bigger problem here. I'm using the wrong Redis key.

**Comment by 0xB10C** [comment]
Thanks @Tomasvrba! Addressed.

File: tools/bbbsupervisor/supervisor/util.go (hunk 4 above)
Comment (0xB10C) [reply tools/bbbsupervisor/supervisor/util.go]: Using `bitcoind:ibd-clearnet` now, not `bitcoind:ibd`.

**Comment by 0xB10C** [comment]
Thank you Tom! 

For context:
I originally planned to have one big PR with a detailed documentation and more context, but had to split it up into two due to interlinked Go Module dependencies for two projects in the same repo. Forgot to add the reasoning here. Will hopefully make more sense in the second PR.

I thought about answering your questions here, but I guess it makes more sense to link to the complete documentation in the second PR.

**Pull request merged**